starlite/cli.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import annotations
+
+import importlib
+import inspect
+from os import getenv
+from pathlib import Path
+from typing import TYPE_CHECKING, Any
+
+import rich
+import rich.tree
+from click import ClickException, argument, confirm, group, option, style
+from rich.console import Console
+from rich.table import Table
+
+from starlite import DefineMiddleware, HTTPRoute, Starlite, WebSocketRoute
+from starlite.utils import is_class_and_subclass
+from starlite.utils.helpers import get_name, unwrap_partial
+
+if TYPE_CHECKING:
+    from starlite.middleware.session.base import ServerSideBackend
+
+console = Console()
+
+
+class _StarliteCLIException(ClickException):
+    def __init__(self, message: str) -> None:
+        super().__init__(style(message, fg="red"))
+
+
+def _autodiscover_app() -> tuple[str, Starlite]:
+    if app_path := getenv("STARLITE_APP"):
+        console.print(f"Using starlite app from env: [bright_blue]{app_path!r}")
```

Comment:
Perhaps we should cap the framework name here to match the rest of the commits?

 `console.print(f"Using Starlite app from env: [bright_blue]{app_path!r}")`